aiida/schedulers/plugins/sge.py

```diff
@@ -279,7 +279,7 @@ def _get_submit_script_header(self, job_tmpl):
             if not isinstance(job_tmpl.job_environment, dict):
                 raise ValueError('If you provide job_environment, it must be a dictionary')
             for key, value in job_tmpl.job_environment.items():
-                lines.append(f'export {key.strip()}={escape_for_bash(value)}')
+                lines.append(f'export {key.strip()}={escape_for_bash(value, job_tmpl.use_double_quotes)}')
```

Comment:
As it stands, we are forcing subclasses to know about this new option and properly call it, or it will effectively be ignored. Seems like a bad design to me. (I realize this is the current state and not something you are introducing with this PR).

I find it weird that this is handled in each subclass. If you look at the code, most `Scheduler` plugins actually do exactly the same, and don't really see how this should be scheduler specific. I think this should simply be moved to a method on the base class and the base class provides a base implementation for `_get_submit_script_header`. Subclasses can then customize only the parts they need.

Reply:
But the syntax of the script file of different schedulers is not the same.  I think only this environment variable part can move to the parent class. Or I think using templates for different schedulers can be another option to implement this. Do you have any better idea in doing this?

Reply:
Yeah, for now we should simply extract the writing of the environment variables. Whipped up a quick fix here: https://github.com/aiidateam/aiida-core/pull/5283

Reply:
The PR is merged, so you can rebase and simply make the change now in one place.